lib/puppet/provider/foreman_resource/rest_v3.rb

```diff
@@ -99,14 +99,16 @@ def success?(response)
   end
 
   def error_message(response)
+    fqdn = URI::parse(resource[:base_url]).host
+
     explanations = {
-      '400' => 'Something is wrong with the data sent to Foreman server',
-      '401' => 'Often this is caused by invalid Oauth credentials',
-      '404' => 'The requested resource was not found',
-      '500' => 'Check /var/log/foreman/production.log on Foreman server for detailed information',
-      '502' => 'The webserver received an invalid response from the backend service. Was Foreman unable to handle the request?',
-      '503' => 'The webserver was unable to reach the backend service. Is foreman.service running?',
-      '504' => 'The webserver timed out waiting for a response from the backend service. Is Foreman under unusually heavy load?'
+      '400' => "Something is wrong with the data sent to Foreman at #{fqdn}",
+      '401' => "Often this is caused by invalid Oauth credentials sent to Foreman at #{fqdn}",
```

Comment:
I'm not sure this change really adds anything. At the CLI we have oauth parameters. Those are wrong and the user needs to correct those. Does this comment help them find that out?

Reply:
The change does clarify that the issue is the Oauth credentials that are being _sent_. So it could dispel the notion that the credentials are wrong somewhere on the Foreman side.

The intent is that the user will compare 1. what does Foreman expect to receive? vs. 2. What am I passing to the CLI?